[v1-007 & v1-009] Added New SignIn page & Responsive Design

NikitaKhadnevich opened the pull request:
This pull includes next changes:
- Added new folders for components;
- Added file-components as like Sign Form( contain sub-components for this Form) , FormikWrapper for this form, MainGrid component etc.
- Created Fokmik and Yup constants and rules for work;
- Deploy other constants and implement in project
- Added styles for components;
- Added basic React Router for work and testing link for this. 
As result we have working form, which can receive User data and work on  requires

hauteballe commented on frontend/src/App.tsx:
```diff
@@ -1,7 +1,22 @@
+    <>
+      <CssBaseline />
+      <Router>
+        <Link to={signIn} style={{ width: '100%', textAlign: 'center', textDecoration: 'none' }}>
```
Please move all the styles inside separate file

NikitaKhadnevich replied:
Its only test link for demo.  In 003 we will replace this Link components.

k-krasilnikova replied:
But could you please move all the styles of it to the styled file?

NikitaKhadnevich replied:
Yup, Sure

hauteballe commented on frontend/src/AppReceiver.tsx:
```diff
@@ -0,0 +1,4 @@
+import SignInContainer from './pages/SignPages/SignInContainer';
+import { signIn, signUp, myCourses, myProfile, requests } from './config/constants/navigationPath';
```
Please use absolute paths here

NikitaKhadnevich replied:
Added

hauteballe commented on frontend/src/components/SignForms/SignInForm/FormikReceiver.tsx:
```diff
@@ -0,0 +1,5 @@
+import signInSchema from "../../../utils/signHelpers/FormikSchemas/signIN_Schema";
+import initSignInvalue from "../../../config/constants/formikConstants";
+import { authLabel,  signUpLabel, afterSignUp } from '../../../config/constants/signConstants';
```
Absolute paths too

NikitaKhadnevich replied:
Fixed

hauteballe commented on frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx:
```diff
@@ -0,0 +1,20 @@
+    initialValues: initSignInvalue,
+    validationSchema: signInSchema,
+    onSubmit: (values, { resetForm }): void => {
+      console.log(values);
```
console.log

NikitaKhadnevich replied:
Removed

hauteballe commented on frontend/src/utils/signHelpers/FormikSchemas/signIN_Schema.tsx:
```diff
@@ -0,0 +1,15 @@
+  requiredPass,
+} from '../signReceiver';
+
+const signInSchema = object().shape({
```
Please move all the validations to the separate folder src/validations (signInValidationSchema)

NikitaKhadnevich replied:
Replaced in scr and renamed

hauteballe commented on frontend/src/utils/signHelpers/signReceiver.tsx:
```diff
@@ -0,0 +1,3 @@
+import { login, signInMinHeight, signInMaxHeight, requiredLogin, requiredPass } from "../../config/constants/signConstants";
```
Absolute path

NikitaKhadnevich replied:
Fixed

hauteballe commented on frontend/src/components/SignForms/SignInForm/SignComponents/EmailField.tsx:
```diff
@@ -0,0 +1,24 @@
+import InputStyled from './styled';
+
+const EmailField = ({ login, touched, onChangeAccum, handleBlur, errors }: EmailProps) => (
+  <InputStyled
```
I think here should be basic custom input from src/components later ( but it's not merged to develop yet )

NikitaKhadnevich replied:
Let's leave it as it is. When 003 will drop I will replace it or fix it otherwise

hauteballe commented on frontend/src/components/SignForms/SignInForm/SignComponents/PasswordField.tsx:
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import TextField from '@mui/material/TextField';
+import { PasswordProps } from '../../SignInterfaces';
+import InputStyled from './styled';
```
Please check all your imports and divide them into 3 groups: libs, absolute paths and relative paths (there must be an empty line between them)

NikitaKhadnevich replied:
Fixed

hauteballe commented on frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx:
```diff
@@ -0,0 +1,33 @@
+          </Typography>
+        </Box>
+        <Box style={{marginTop: '20px'}}>
+          <Typography variant="h5" style={{color: 'gray'}}>
```
styles

NikitaKhadnevich replied:
Replaced in styled.tsx

hauteballe commented on frontend/src/pages/SignPages/SignReceiver.tsx:
```diff
@@ -0,0 +1,5 @@
+import FormikInContainer from '../../components/SignForms/SignInForm/FormikInContainer';
```
absolute paths

NikitaKhadnevich replied:
Fixed

hauteballe reviewed (changes requested):
Some changes required

k-krasilnikova commented on frontend/src/AppReceiver.tsx:
```diff
@@ -0,0 +1,4 @@
+import SignInContainer from './pages/SignPages/SignInContainer';
```
Do we really need this file? Let's remove it

NikitaKhadnevich replied:
I replace it and added other files. Maybe will this file will be receiver for import? For example, path, components, other constants

k-krasilnikova commented on frontend/src/components/SignForms/SignInForm/FormikInContainer.tsx:
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Remove please this file and move all the logic inside SignIn container
Rename please all the path to the: `frontend/src/pages/SignIn/SignInContainer.tsx` and `frontend/src/pages/SignIn/SignIn.tsx`

NikitaKhadnevich replied:
_**Remove please this file and move all the logic inside SignIn container**_ 
Lets see, in future I will use useQuery and i want to add this into FormikContainer - now SignIn already include over 70n lines, and its almost all only Sign-form describe. If i will add logic yet it will be not fine, this file wll include over then 120-150 lines. Maybe i will rename FormikContainer to SignContainer and

**_Rename please all the path to the: frontend/src/pages/SignIn/SignInContainer.tsx and frontend/src/pages/SignIn/SignIn.tsx_**
will rename it as SignPage.tsx and also will fix path.

hauteballe replied:
You should have SignIn folder with SignIn.tsx and SignInContainer.tsx inside it. Now we have too much extra folders and components here

NikitaKhadnevich replied:
Ok, I see that folders-hell))) replaced and fixed paths and remove some files

hauteballe replied:
SignInForm is not a reusable component, so it rather should be in pages/SignIn folder

k-krasilnikova commented on frontend/src/components/SignForms/SignInForm/SignComponents/SignLabel.tsx:
```diff
@@ -0,0 +1,13 @@
+import { Typography } from '@mui/material';
+import { authLabel } from '../FormikReceiver';
+
+const SignLabel = (): any => (
```
Please don't use `any` type anywhere, it's not safe enough

NikitaKhadnevich replied:
Already fixed

k-krasilnikova commented on frontend/src/components/SignForms/SignInForm/SignComponents/SignLabel.tsx:
```diff
@@ -0,0 +1,13 @@
+import { Typography } from '@mui/material';
+import { authLabel } from '../FormikReceiver';
+
+const SignLabel = (): any => (
```
I'm not sure that it should be a separated component

k-krasilnikova commented on frontend/src/components/SignForms/SignInForm/SignIn.tsx:
```diff
@@ -0,0 +1,70 @@
+import EmailField from './SignComponents/EmailField';
+import PasswordField from './SignComponents/PasswordField';
+import SignLabel from './SignComponents/SignLabel';
+import { SignInFormik } from '../SignInterfaces';
```
Remove this file and change the structure of SignIn page

NikitaKhadnevich replied:
Done

k-krasilnikova commented on frontend/src/components/SignForms/SignInForm/SignIn.tsx:
```diff
@@ -0,0 +1,70 @@
+    handleSubmit,
+  } = formik;
+
+  const onChangeAccum = (name: boolean, e: string): any => {
```
any?
What is Accum?
What is true and false sequences? Do we really need this method?

NikitaKhadnevich replied:
renamed onChangeAccum => onChangeHandler.

NikitaKhadnevich replied:
@k-krasilnikova renamed "onChangeAccum => onChangeHandler", any replaced. On my opinion its elegant approach for work with input data and errors if input data wrong. Lets see, Usually when the User starting enter the data called onChange and after we will handle result and show in additional filed under input error info or other info. "_<input ..../> + isError <p>Error</p>_". But this meta work otherwise - we will bind onChangeHandler and after every User tap will call onChangeHandler after/im call onChange and finally calling setFieldTouched. This method has 3 arg - name, true, false. **Name** - its input name for identificate. **True** - its isTouched status(if user starting enter the data isTouched anyway will be true) and last arg - **False** its helper which explicitly prevent/skip validation for every tap. Magic in "helperText" prop - helperText={touched.fieldname && errors.fieldname). each keyboard click in input will call onChangeHandler and based on input data and will observe - if smth wrong in touched(touched will be is not empty)  immediately helperText show to User wrong if all right - touched will be empty. We dont need to create additional fields for work as like paragraph and. also there are not problem with fields and code will be clear - this method fill only 3 lines. fuuh)) Lets save it?)

k-krasilnikova commented on frontend/src/components/SignForms/SignInterfaces.tsx:
```diff
@@ -0,0 +1,47 @@
+interface SignInFormik {
+  formik: {
+    errors: {
+      login?: string | any,
+      password?: string | number
+    },
+    touched: {
+      login?: string | any,
+      password?: string | any
+    },
+    handleChange: (e: any) => any,
+    isValid: boolean,
+    setFieldTouched: (name: any, value: any, isValidate?: any) => any,
+    handleBlur: (e: any) => any,
+    handleSubmit: (e: any) => any
+    values: { 
+      login: string,
+      password: string | number
+    },
```
Remove all the `any` types

NikitaKhadnevich replied:
Removed wherever possible.

k-krasilnikova commented on frontend/src/components/SignForms/SignPresenter/SignPresenter.tsx:
```diff
@@ -0,0 +1,21 @@
+import { authGreetings, techLogIn, techArtLabel } from './SignPresReceiver';
+import { SignPresenterBox, PresenterWrapper, SighTechArtLabel, SighPresentSubtext } from './styled';
+
+const SignPresenter = (): any => {
+  return (
```
any?
There is no need to use `return` statement

NikitaKhadnevich replied:
Fixed

k-krasilnikova commented on frontend/src/config/constants/formikConstants.tsx:
```diff
@@ -0,0 +1,13 @@
+interface FormikInit {
```
wht if i ll wrt cmmnts th sm wy? Rename to something more clear please

NikitaKhadnevich replied:
Renamed

k-krasilnikova commented on frontend/src/config/constants/navigationPath.tsx:
```diff
@@ -0,0 +1,10 @@
+const path = {
+  signIn: 'signin',
+  signUp: 'signup',
+  myCourses: 'mycourses',
+  myProfile: 'profile',
+  requests: 'requests',
+};
+
+const { signIn, signUp, myCourses, myProfile, requests } = path;
+export { signIn, signUp, myCourses, myProfile, requests };
```
What is it? We already have some routing in v1-003 brunch. Reuse them please

NikitaKhadnevich replied:
I had created it before merge v003 into develop branch

NikitaKhadnevich replied:
Fixed it add - signIn into this file

k-krasilnikova commented on frontend/src/pages/SignPages/SignInContainer.tsx:
```diff
@@ -0,0 +1,22 @@
+
+import { SignMain, SignMainGrid, SignPresGrid, SignFormGrid } from './styled';
+
+const SignInContainer = (): any => {
+  return (
+    <SignMain>
```
any? return?

NikitaKhadnevich replied:
Fixed

k-krasilnikova commented on frontend/src/pages/SignPages/SignInContainer.tsx:
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+import { FormikInContainer, SignPresenter } from './SignReceiver';
+
```
Unexpected empty line

NikitaKhadnevich replied:
Fixed

k-krasilnikova commented on frontend/src/themeSettings.ts:
```diff
@@ -0,0 +1,30 @@
+import { createTheme } from '@mui/material';
+
+const colors = {
```
uppercase?

NikitaKhadnevich replied:
Fixed

k-krasilnikova reviewed (changes requested):
Great job, but still some changes needed

hauteballe commented on backend/package.json:
```diff
@@ -32,6 +32,7 @@
         "express": "^4.17.1",
-        "jest": "^27.4.5"
+        "jest": "^27.4.5",
+        "tsconfig-paths": "^3.12.0"
```
Why did you use "tsconfig-paths" lib? Absolute paths are already working with 'baseUrl' setting
Please remove it

NikitaKhadnevich replied:
I dont fix this file - its merge from develop branch before

hauteballe commented on backend/tsconfig.json:
```diff
@@ -8,6 +8,17 @@
+    "include": ["./src/**/*.ts"],
+    "exclude": ["node_modules", "dist"],
+    "ts-node": {
+      "require": ["tsconfig-paths/register"]
```
What is the purpose? Please change it according to the tsconfig.json on develop branch

NikitaKhadnevich replied:
I dont fix this file - its merge from develop branch before

hauteballe commented on frontend/.env:
```diff
@@ -1,5 +0,0 @@
-
-BROWSER=none
-PORT=3000
-PUBLIC_URL=/
```
Why did you remove this part? Have you discussed these changes with someone?

NikitaKhadnevich replied:
oh yes( I tested SPA and forgot return this settings.I will correct its immediately.

hauteballe commented on frontend/src/components/SignForms/SignInForm/styled.tsx:
```diff
@@ -0,0 +1,42 @@
+});
+
+const ItemsBox = styled(Box)({
+  margin: '0px !important',
```
Do we really need to use important directive in all styled.ts files?

NikitaKhadnevich replied:
yup, because basic material ui styles covering margins/paddings and some styles. Without this in future there may be errors in deploy.

hauteballe commented on frontend/src/pages/SignPages/SignReceiver.tsx:
```diff
@@ -0,0 +1,5 @@
+import FormikInContainer from 'components/SignForms/SignInForm/FormikInContainer';
+import SignPresenter from 'components/SignForms/SignPresenter/SignPresenter';
```
Why do you use SignPages and SignForms naming if there is only one page and form?

NikitaKhadnevich replied:
I renamed/deleted it

hauteballe reviewed (changes requested):
And please fix all linter issues
![screen](https://user-images.githubusercontent.com/44285339/147078430-c8b41feb-c313-448c-a6ac-5a02a33e3f9f.jpg)

NikitaKhadnevich commented:
This commit also include **`v1-009 feature`**. For this i had created:
1. Custom resolutions as like in conditions (`320px, 425px, 768px, 1024px, 1440px+`);
2. Added and fix basic MIU styles in `styled.tsx` files;
3. Created `@media` for responsible design in styled files
4. Tested and implement its^^^.
If `v1-009 feature` have to have separate pull/commit, i will overwrite its. Maybe we will have new SPA design in future.

NikitaKhadnevich commented:
_**This pull request includes:**_
1. Refactored lines and folders;
2. Rewrote, renamed and united / split some components;
3. Added more clear methods and functions in containers;
4. Created new and replaced old TS types
5. "Drew" Styles for Responsive design and new resolutions
6. other fix
